### PR TITLE
fix: bump `SelectedNetworkController` to fix issue where selected network domain is incorrectly set

### PR DESCRIPTION
**Pull request by adonesky1**
## **Description**

Pulls [this fix](https://github.com/MetaMask/core/pull/4388) which adds a guard to the `setNetworkClientIdForDomain` method on the `SelectedNetworkController` to no longer add domains to domains state (nor create a selected network proxy for them) unless the `useRequestQueuePreference` flag is true 

Bumping the `@metamask/selected-network-controller` version required [bumping peer dependencies](https://github.com/MetaMask/core/blob/main/packages/selected-network-controller/CHANGELOG.md#changed):
 - `@metamask/network-controller` to v19.0.0
    - An existing patch was repointed to this version.
 - `@metamask/permission-controller` to v10.0.0
    -[ This required bumping its own peerDep](https://github.com/MetaMask/core/blob/main/packages/selected-network-controller/CHANGELOG.md#changed) `@metamask/approval-controller` to v7.0.0

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25106?quickstart=1)

## **Related issues**

See: https://consensys.slack.com/archives/C1L7H42BT/p1717512150509719

## **Manual testing steps**

1. Open settings -> experimental and toggle of the `Select networks for each site` setting
2. Go to any site
3. Open the console and execute:
```
await window.ethereum.request({
  "method": "eth_chainId",
  "params": []
});
```
4. See that it matches the globally selected network
5. Manually change the network with the network switcher
6. Go back to the site and execute the same `eth_chainId` script
7. See that the result has changed to the chainId you switched to
8. Now execute
```
await window.ethereum.request({
  "method": "wallet_requestPermissions",
  "params": [
    {
      "eth_accounts": {}
    }
  ]
});
```
and connect an account to the site
9. Repeat steps 1 - 7
10. The `eth_chainId` results should still match the globally selected network

### **Before**

https://github.com/MetaMask/metamask-extension/assets/34557516/ff2f68ff-6ec8-495f-96c1-e938973ce59b

### **After**

https://github.com/MetaMask/metamask-extension/assets/34557516/6061f4e2-faab-48a5-8c55-3b2a4e3aa9f2

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -252,7 +252,10 @@
     "sucrase@npm:3.34.0": "^3.35.0",
     "@expo/config/glob": "^10.3.10",
     "@expo/config-plugins/glob": "^10.3.10",
-    "@metamask/network-controller": "patch:@metamask/network-controller@npm%3A18.1.2#~/.yarn/patches/@metamask-network-controller-npm-18.1.2-1bcb8d8610.patch"
+    "@metamask/network-controller@npm:^19.0.0": "patch:@metamask/network-controller@npm%3A19.0.0#~/.yarn/patches/@metamask-network-controller-npm-19.0.0-a5e0d1fe14.patch",
+    "@metamask/network-controller@npm:^17.0.0": "patch:@metamask/network-controller@npm%3A19.0.0#~/.yarn/patches/@metamask-network-controller-npm-19.0.0-a5e0d1fe14.patch",
+    "@metamask/network-controller@npm:^18.1.0": "patch:@metamask/network-controller@npm%3A19.0.0#~/.yarn/patches/@metamask-network-controller-npm-19.0.0-a5e0d1fe14.patch",
+    "@metamask/network-controller@npm:^18.1.2": "patch:@metamask/network-controller@npm%3A19.0.0#~/.yarn/patches/@metamask-network-controller-npm-19.0.0-a5e0d1fe14.patch"
```
Comment (Gudahtt) [inline package.json:258]: I know yarn adds these entries automatically when you generate patches, but in this case I think we only need to patch the one we're directly using. The rest are just used for type reasons, by dependencies.
```suggestion
    "@metamask/network-controller@npm:^19.0.0": "patch:@metamask/network-controller@npm%3A19.0.0#~/.yarn/patches/@metamask-network-controller-npm-19.0.0-a5e0d1fe14.patch"
```

Comment (adonesky1) [reply package.json:258]: Yep I meant to remove those.

**Comment by adonesky1** [comment]
@metamaskbot update-policies

**Comment by adonesky1** [comment]
@metamaskbot update-policies

**Review by mcmire — APPROVED** [review]
LGTM.

**Review by jiexi — APPROVED** [review]

**Pull request merged**